pyteal/ast/abi/tuple.py

```diff
@@ -128,7 +129,7 @@ def indexTuple(
         offset += typeBefore.byte_length_static()
 
     valueType = valueTypes[index]
-    if not valueType.has_same_type_as(output):
+    if not valueType.has_same_type_as(output) or not isinstance(output, Type):
```

Comment:
nit: better add some comments to explain this condition checks: `output` is an instance of `Type`, while `output` has same type as `valueType`.
My apologies for asking for more explanation, but we are dealing with both types and instantiations, which make things moar complex...

Reply:
This comment is now outdated